etc/subsystem_manifest/main.go

```diff
@@ -0,0 +1,141 @@
+)
+
+const (
+	viamServer = "viam-server"
+	linuxAmd64 = "linux/amd64"
+	linuxArm64 = "linux/arm64"
+)
+
+type dumpedResourceRegistration struct {
+	API             string             `json:"api"`
+	Model           string             `json:"model"`
+	AttributeSchema *jsonschema.Schema `json:"attribute_schema,omitempty"`
+}
+
+type viamServerMetadata struct {
+	ResourceRegistrations []dumpedResourceRegistration `json:"resource_registrations"`
+}
+
+type substemManifest struct {
+	Subsystem  string              `json:"subsystem"`
+	Version    string              `json:"version"`
+	Platform   string              `json:"platform"`
+	ObjectPath string              `json:"object-path"`
+	Sha256     string              `json:"sha256"`
+	Metadata   *viamServerMetadata `json:"metadata,omitempty"`
+}
+
+func main() {
+	subsystem := flag.String("subsystem", viamServer, "subsystem type") // default to viam-server
+	binaryPath := flag.String("binary-path", "", "path to subsystem binary")
+	objectPath := flag.String("object-path", "", "path where this binary will be stored in gcs")
+	version := flag.String("version", "", "version")
+	arch := flag.String("arch", "", "arch (result of uname -m) ex: x86_64")
+
+	flag.Parse()
+
+	if *binaryPath == "" || *objectPath == "" || *arch == "" || *version == "" {
+		log.Fatal("binary-path, arch, version, and url are required arguments")
+	}
+
+	binarySha, err := sha256sum(*binaryPath)
+	if err != nil {
+		log.Fatalf("failed to calculate binary sha: %v", err)
+	}
+	metadata, err := getViamServerMetadata(*binaryPath)
+	if err != nil {
+		log.Fatalf("failed to get viam-server metadata: %v", err)
+	}
+	platform, err := osArchToViamPlatform(*arch)
+	if err != nil {
+		log.Fatalf("failed to get platform: %v", err)
+	}
+
+	manifest := substemManifest{
+		Subsystem:  *subsystem,
+		Version:    strings.TrimPrefix(*version, "v"),
+		Platform:   platform,
+		ObjectPath: *objectPath,
+		Sha256:     binarySha,
+		Metadata:   metadata,
+	}
+
+	// marshall and output the manifest to stdout
+	jsonResult, err := json.MarshalIndent(manifest, "", "\t")
+	if err != nil {
+		log.Fatalf("failed to marshall json result %v", err)
+	}
+
+	if _, err := os.Stdout.Write(jsonResult); err != nil {
+		log.Fatalf("failed to print result %v", err)
+	}
+}
+
+func getViamServerMetadata(path string) (*viamServerMetadata, error) {
+	output := new(bytes.Buffer)
+	processConfig := pexec.ProcessConfig{
+		Name:      path,
+		Args:      []string{"--dump-resources"},
+		OneShot:   true,
+		Log:       false,
+		LogWriter: output,
+	}
+	proc := pexec.NewManagedProcess(processConfig, zap.NewNop().Sugar().Named("blank"))
+	if err := proc.Start(context.Background()); err != nil {
+		return nil, err
+	}
+	dumpedResourceRegistrations := []dumpedResourceRegistration{}
+	if err := json.Unmarshal(output.Bytes(), &dumpedResourceRegistrations); err != nil {
+		return nil, err
+	}
+	return &viamServerMetadata{
+		ResourceRegistrations: dumpedResourceRegistrations,
+	}, nil
+}
```

Comment:
Note that we don't simply import `rdk/components/register` and `rdk/services/register` in this file because then this script would have to be run with exactly the same build-flags as the viam-server which seems error-prone and brittle